contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -47,448 +43,251 @@ contract FraudProofFacet is StateChannelCommon {
+            return address(0);
+        }
+
+        if(block1.stateSnapshotHash != block2.stateSnapshotHash && block1.previousBlockHash != block2.previousBlockHash) {
+            return address(0);
+        }
+        
+        address signer1 = StateChannelUtilLibrary.retriveSignerAddress(
+            blockDoubleSignProof.block1.encodedBlock,
+            blockDoubleSignProof.block1.signature
+        );
+        address signer2 = StateChannelUtilLibrary.retriveSignerAddress(
+            blockDoubleSignProof.block2.encodedBlock,
+            blockDoubleSignProof.block2.signature
+        );
+        if(signer1 != signer2) {
+            return address(0);
+        }
+        return signer1; 
     }
 
-    function _handleBlockEmptyBlock(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
-        // Implementation for handling BlockEmptyBlock proof
-        return address(0); // Replace with actual logic
+    function _handleBlockEmptyBlock(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) pure internal returns (address) {
+        BlockEmptyProof memory blockEmptyProof = abi.decode(encodedProof, (BlockEmptyProof));
+        Block memory fraudBlock = abi.decode(blockEmptyProof.emptyBlock.encodedBlock, (Block));
+
+        if(fraudProofVerificationContext.channelId != fraudBlock.transaction.header.channelId) {
+            return address(0);
+        }
+        if(fraudBlock.transaction.header.transactionCnt != uint(0)) {
+            return address(0);
+        }
+        address signer = StateChannelUtilLibrary.retriveSignerAddress(
+            blockEmptyProof.emptyBlock.encodedBlock,
+            blockEmptyProof.emptyBlock.signature
+        );
+        return signer;
     }
 
     function _handleBlockInvalidStateTransition(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
-        // Implementation for handling BlockInvalidStateTransition proof
-        return address(0); // Replace with actual logic
+        BlockInvalidStateTransitionProof memory blockInvalidSTProof = abi.decode(encodedProof, (BlockInvalidStateTransitionProof));
+        Block memory fraudBlock = abi.decode(blockInvalidSTProof.invalidBlock.encodedBlock, (Block));
+        
+        if(fraudProofVerificationContext.channelId != fraudBlock.transaction.header.channelId) {
+            return address(0);
+        }
+
+        (bool isTransitionValid, bytes memory encodedModifiedState) = AStateChannelManagerProxy(address(this)).executeStateTransitionOnState(
+            fraudProofVerificationContext.channelId,
+            blockInvalidSTProof.encodedLatestState,
+            fraudBlock.transaction
+        );
+
+        if (isTransitionValid) {
+            return address(0);
+        } 
+
+        if(fraudBlock.stateSnapshotHash == keccak256(encodedModifiedState)){
```

Comment:
This is also not correct - you can't compare hash of a snapshot to a hash of a state(bytes) - you need to reconstruct the entire snapshot and compare it to the commitment - see how I constructed the snapshot during auditing 

Assume previous snapshot is correct and based on that do the state transition and calculate the output snapshot and compare. 
Deposits stay the same since this can only be changed on-chain, and if exits happen the state transition function needs to return exitChannels [ ]